tests/python/rest_api/test_cloud_storages.py

```diff
@@ -543,6 +617,46 @@ def _test_get_cloud_storage_content(
                     FileInfo(mime_type="unknown", name="manifest_2.jsonl", type="REG"),
                 ],
             ),
+            (
+                # [v2] list real bucket content when default bucket prefix
+                # is set to template from which the files should start
+                SUPPORTED_VERSIONS.V2,
+                None,
+                None,
+                "sub/demo",
+                None,
+                [
+                    FileInfo(mime_type="unknown", name="demo_manifest.jsonl", type="REG"),
+                ],
+            ),
+            (
+                SUPPORTED_VERSIONS.V2,  # [v2] list real bucket content when specified prefix is stricter than default bucket prefix
+                None,
+                "sub/image_case_65_1",
+                "sub/image_case",
+                None,
+                [
+                    FileInfo(mime_type="image", name="image_case_65_1.png", type="REG"),
+                ],
+            ),
+            (
+                SUPPORTED_VERSIONS.V2,  # [v2] list real bucket content when default bucket prefix is stricter than specified prefix
+                None,
+                "sub/image_case",
+                "sub/image_case_65_1",
+                None,
+                [
+                    FileInfo(mime_type="image", name="image_case_65_1.png", type="REG"),
+                ],
+            ),
+            (
+                SUPPORTED_VERSIONS.V2,  # [v2] list real bucket content when default bucket prefix and specified prefix have no intersection
+                None,
+                "sub/image_case_65_1",
+                "sub/image_case_65_2",
+                None,
+                [],
+            ),
```

Comment:
What about this case? https://github.com/opencv/cvat/blob/cdd4040a737dc2646f5f139dcbe7d9929fd319a5/cvat/apps/engine/cloud_provider.py#L269-L272

Any tests for it? I only see test cases with regular files.